cpp/src/io/parquet/page_data.cu

```diff
@@ -90,6 +95,13 @@ struct page_state_s {
   const uint8_t* lvl_start[NUM_LEVEL_TYPES];  // [def,rep]
   int32_t lvl_count[NUM_LEVEL_TYPES];         // how many of each of the streams we've decoded
   int32_t row_index_lower_bound;              // lower bound of row indices we should process
+
+  // a shared-memory cache of frequently used data when decoding. The source of this data is
+  // normally stored in global memory which can yield poor performance. So, when possible
+  // we copy that info here prior to decoding
+  PageNestingDecodeInfo nesting_decode_cache[max_cacheable_nesting_decode_info];
+  // points to either nesting_decode above when possible, or to the global source otherwise
+  PageNestingDecodeInfo* pndi;
```

Comment:
Could this be named something like `decode_info` or `page_nesting_cache`? Just real words that indicate the purpose, since `pndi` is just a type abbreviation that loses its referent when used with `auto` or in untyped attribute accessors.

As a bonus: The comment `points to either...` is helpful but I would like it even more if we could encode that idea into the name while avoiding Hungarian notation? (Maybe not possible)

Reply:
The thing I don't like about `decode_info` is that this is more specific than that.  It's decode info related to the nesting of the columns (the `page_state_s` struct itself is really the `decode_info`).  I used `pndi` to make usage of it less crazy to read.  Maybe `nesting_info`?

Reply:
`nesting_info` +1